app/model.py

```diff
@@ -0,0 +1,114 @@
+from db_connect import TrackerDB
+from psycopg2.extras import RealDictCursor
+from passlib.hash import pbkdf2_sha256 as sha256
+import psycopg2
+
+db = TrackerDB()
+
+
+class User():
+    def __init__(self, username, email, password):
+        self.username = username
+        self.email = email
+        self.password = password
+
+    def create_user(self):
+        db.cur.execute("""INSERT INTO users(username, email, password)
+                VALUES(%s,%s,%s)""",
+                       (self.username, self.email,
+                        generate_hash(self.password),))
+        db.conn.commit()
+
+
+def generate_hash(password):
+    return sha256.hash(password)
+
+
+def verify_hash(password, hash):
+    return sha256.verify(password, hash)
+
+
+def get_users():
+    db.cur.execute("SELECT * FROM users")
+    db.conn.commit()
+    users = db.cur.fetchall()
+    return users
+
+
+def get_user_by_id(user_id):
+    db.cur.execute("SELECT * FROM users WHERE id = (%s)", (user_id,))
+    db.conn.commit()
+    users = db.cur.fetchall()
+    return users
+
+
+def get_user(username):
+    db.cur.execute("SELECT * FROM users WHERE username = (%s);", (username,))
+    db.conn.commit()
+    user = db.cur.fetchone()
+    return user
+
+
+class UserRequest():
```

Comment:
you are required to add a field for the request status that is only editable by the admin.